File: ruby/bob/bob.rb
# The Bob Class
class Bob
  def self.hey(remark)
    remark = remark.gsub(/\s/, '')
    if remark.empty?
      'Fine. Be that way!'
    elsif !remark.gsub(/[^a-zA-Z]/, '').empty? && remark == remark.upcase
      'Whoa, chill out!'
    elsif remark[-1] == '?'
      'Sure.'
    else
      'Whatever.'
    end
  end

  module BookKeeping
    VERSION = 1
  end
end

File: javascript/bob/bob.js
class Bob {
  hey(str) {
    str = str.trim();
    switch(true) {
      case (this.isNotSyaingAnything(str)):
        return 'Fine. Be that way!';
      case (this.isYellingAtMe(str)):
        return 'Whoa, chill out!'
      case (this.isQuestioningMe(str)):
        return 'Sure.'
      default:
        return 'Whatever.'
    }
  }

  isNotSyaingAnything(str) {
    return str.replace(/\s/g, '') === '';
  }

  isYellingAtMe(str) {
    if (!str.replace(/[^a-zA-Z]/g, '')) { return false; }
    return str.toUpperCase() === str;
  }

  isQuestioningMe(str) {
    return str.slice(-1) === '?';
  }
}

module.exports = Bob;